fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -915,8 +915,21 @@ class BrowserRobot {
     fun fillPdfForm(name: String) {
         // Set PDF form text for the text box
         itemWithResId("pdfjs_internal_id_10R").setText(name)
-        // Click PDF form check box
-        itemWithResId("pdfjs_internal_id_11R").click()
+        mDevice.waitForWindowUpdate(packageName, waitingTime)
+        if (
+            itemWithResId("pdfjs_internal_id_11R").exists().not() &&
+            mDevice
+                .executeShellCommand("dumpsys input_method | grep mInputShown")
+                .contains("mInputShown=true")
+        ) {
+            // Close the keyboard
+            mDevice.pressBack()
+            // Click PDF form check box
+            itemWithResId("pdfjs_internal_id_11R").click()
```

Comment:
This can be taken out of the if statement, you can leave inside only the part where you are dismissing the keyboard.

Reply:
Done !